Explorer/Assets/Scripts/ECS/Unity/PrimitiveRenderer/Systems/InstantiatePrimitiveRenderingSystem.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using CrdtEcsBridge.Components.Transform;
+using DCL.ECSComponents;
+using ECS.Abstract;
+using ECS.ComponentsPooling;
+using ECS.Unity.Groups;
+using ECS.Unity.PrimitiveRenderer.Components;
+using ECS.Unity.Transforms.Components;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ECS.Unity.PrimitiveRenderer.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    public partial class InstantiatePrimitiveRenderingSystem : BaseUnityLoopSystem
+    {
+        private readonly IComponentPool<MeshRenderer> rendererPoolRegistry;
+        private readonly IComponentPool<Mesh> meshPoolRegistry;
+
+        private readonly Dictionary<PBMeshRenderer.MeshOneofCase, ISetupMesh> SETUP_MESH_LOGIC = new ()
+        {
+            { PBMeshRenderer.MeshOneofCase.Box, new SetupBoxMesh() },
+            { PBMeshRenderer.MeshOneofCase.Sphere, new SetupSphereMesh() },
+            { PBMeshRenderer.MeshOneofCase.Cylinder, new SetupCylinder() },
+            { PBMeshRenderer.MeshOneofCase.Plane, new SetupPlaneMesh() },
+        };
+
+        private readonly Dictionary<PBMeshRenderer.MeshOneofCase, ISetupMesh> setupMeshCases;
+
+        internal InstantiatePrimitiveRenderingSystem(World world, IComponentPoolsRegistry poolsRegistry,
+            Dictionary<PBMeshRenderer.MeshOneofCase, ISetupMesh> setupMeshCases = null) : base(world)
+        {
+            this.setupMeshCases = setupMeshCases ?? SETUP_MESH_LOGIC;
+
+            rendererPoolRegistry = poolsRegistry.GetReferenceTypePool<MeshRenderer>();
+            meshPoolRegistry = poolsRegistry.GetReferenceTypePool<Mesh>();
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateNonExistingRendererQuery(World);
+            TrySetupExistingRendererQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBMeshRenderer), typeof(SDKTransform), typeof(TransformComponent))]
+        [None(typeof(PrimitiveRendererComponent))]
+        private void InstantiateNonExistingRenderer(in Entity entity, ref PBMeshRenderer sdkComponent, ref TransformComponent transform)
+        {
+            var rendererComponent = new PrimitiveRendererComponent();
```

Comment:
In general, it's fine but I would like to warn you that `PrimitiveRendererComponent` and `PrimitiveMeshComponent` always come hand in hand, it's a signal that they should be a part of the same component. You still can have different structures if you want (composition)